src/assignments/tech-jobs-oo.rst

```diff
@@ -0,0 +1,558 @@
+   b. Which constructors are the same in ALL FOUR classes?
+   c. Which custom methods are identical in ALL of the classes?
+
+#. In ``JobField``, declare each of the common fields.
+#. Code the constructors.
+#. Add in the custom methods.
+#. Finally, to prevent the creation of a ``JobField`` object, make this class
+   *abstract*.
+
+Extend ``JobField`` into ``Employer``
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Now that you have the common code located in the ``JobField`` file, we can
+modify the other classes to reference this shared code. Let's begin with
+``Employer``.
+
+#. Modify line 5 to *extend* the ``JobField`` class into ``Employer``.
+#. Next, remove any code in ``Employer`` that matches code from ``JobField``
+   (e.g. the ``Id``, ``Value``, and ``nextId`` fields are shared).
+#. Remove any of the methods that are identical.
+#. The empty constructor is shared, but not the second. Replace the two
+   constructors with the following:
+
+   .. sourcecode:: csharp
+      :lineno-start: 7
+
+      public Employer(string value) : base() 
+      {
+         Value = value;
+      }
+
+   The ``:`` and ``base`` keywords link the ``JobField`` and
+   ``Employer`` classes.
+#. Rerun your unit tests to verify your refactored code.
+
+Finish DRYing Your Code
+^^^^^^^^^^^^^^^^^^^^^^^
+
+#. Repeat the process above for the ``Location``, ``CoreCompetency``, and
+   ``PositionType`` classes.
+#. Rerun your unit tests to verify that your classes and methods still work.
+
+.. admonition:: Tip
+
+   You know you need to do this, but here is the reminder anyway. Save, commit,
+   and push your work to GitHub.
+
+Sanity Check
+-------------
+
+Once you finish all of the tasks outlined above, all that remains is to check
+the console display.
+
+Sally has provided some commented-out code in ``Program.cs`` that prints out a small
+list of ``Job`` objects. Go ahead and activate this code and run it.
+Properly done, your output should look something like:
+
+::
+
+   ID: 1
+   Name: Product tester
+   Employer: ACME
+   Location: Desert
+   Position Type: Quality control
+   Core Competency: Persistence
+
+
+   ID: 2
+   Name: Web Developer
+   Employer: LaunchCode
+   Location: St. Louis
+   Position Type: Front-end developer
+   Core Competency: JavaScript
+
+
+   ID: 3
+   Name: Ice cream tester
+   Employer: Data not available
+   Location: Home
+   Position Type: UX
+   Core Competency: Tasting ability
+
+Excellent! You successfully shifted the old console app into a more useful
+object oriented configuration.
+
+Now that the new structure is ready, another team member can refactor the
+import and display methods from the previous assignment to use the new classes. Once these are ready, our
+team will refine the search features and move the app online to provide a
+better user interface.
+
+How to Submit
+--------------
+
+To turn in your assignment and get credit, follow the
+:ref:`submission instructions <how-to-submit-work>`.
```

Comment:
on this submission instructions page, add a link to this assignment too

Reply:
Fixed!